### PR TITLE
feat(Flexbox layout): Provides flexbox layout 

### LINKED ISSUE #957: Introduce new flexbox layout
This is related to #886 
Our currently toolbar layout needs to be updated to be a component which leaves us with a need for a more generic type of layout to meet the needs that our current toolbar layout was fulfilling. 
This should be a flexbox layout with modifiers for spacing.
We still need to investigate the overall requirements and use cases for this layout to ensure it is flexible enough.

### PR DESCRIPTION
**Pull request by mattnolting**
Handing off work, heading into paternity leave

There are several utility modifiers included, some may need to be responsive, some may not and some may need to be added

Closes #957

### CODE REVIEW

**Comment by mcoker** [comment]
You can ignore `flex-ba.scss`, that's something Matt Nolting was working from and I'll leave it in case he wants to continue when he gets back.

File: src/patternfly/sass-utilities/scss-variables.scss [hunk 1]
```diff
@@ -144,6 +144,11 @@ $pf-global--breakpoint--md: 768px !default;
 $pf-global--breakpoint--lg: 992px !default;
 $pf-global--breakpoint--xl: 1200px !default;
 
+// Device breakpoints
```
Comment (matthewcarleton) [inline src/patternfly/sass-utilities/scss-variables.scss]: Do we need to add these new breakpoints. Can't we just use what we already have and have a min/max breakpoint for the mobile vs tablet and then use the xl breakpoint?

File: .gitignore [hunk 2]
```diff
@@ -3,6 +3,7 @@ node_modules/
 test/scenario_tests/
 test/results/
 .sass-cache/
+.vscode/
```
[the remark this answers is not in the document]
Comment (mcoker) [reply .gitignore]: Yeah I think that's fine. It's so you can setup workspace specific settings. Looks like Matt does that.

File: src/patternfly/layouts/Flex/examples/flex-alignment-example.hbs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    {{#> example example--modifier="blue"}}4{{/example}}
+    {{#> example example--modifier="blue"}}5{{/example}}
+  {{/flex}}
+  {{#> flex newcontext flex--modifier="pf-m-align-right"}}
+    {{#> example example--modifier="red"}}6{{/example}}
+    {{#> example example--modifier="red"}}7{{/example}}
+  {{/flex}}
+{{/flex}}
+
+{{#> example-title}}A better alignment method, <code>.pf-m-flex-grow</code> + <code>.pf-m-nowrap</code>.{{/example-title}}
```
[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/layouts/Flex/examples/flex-alignment-example.hbs]: good call. changed to "A more differenter method"

[the remark this answers is not in the document]
Comment (matthewcarleton) [reply src/patternfly/layouts/Flex/examples/flex-alignment-example.hbs]: ah can you drop the one after it too, sorry should have pointed that one out too.

[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/layouts/Flex/examples/flex-alignment-example.hbs]: roger that, updated.

**Pull request merged**